pyairtable/request_strategies/retrying.py

```diff
@@ -0,0 +1,252 @@
+from typing import Any, Optional, overload
```

Comment:
Is there a way to simplify this (module)? 
Given the size of the library, this is sizable addition and I am concerned about maintaining a lot of complex logic in this module, some of which I am still trying to wrap my head around